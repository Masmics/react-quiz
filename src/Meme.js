import React, { PureComponent } from 'react';
import PropType from 'prop-types';
import TextSelector from '';
import Display from '';

export default class Meme extends PureComponent {
  state = {
    header: 'Meme Generator',
    image: 'https://images.ecosia.org/An2PbD_aWxo5_5lkuaiDNB7sl-s=/0x390/smart/http%3A%2F%2Fi0.kym-cdn.com%2Fphotos%2Fimages%2Foriginal%2F001%2F178%2F872%2F3f4.jpg',
    footerText: ''
  }
  
  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }
  
  render() {
    const { header, image, footerText } = this.state;
    return (
      <>
        <TextSelector
          footerText={footerText}
          handleChange={this.handleChange}
        />
        <Display
          header={header}
          image={image}
          footerText={footerText}
        />
      </>
    );
  }
}


// Container component
// contains state
// Uses the other components
