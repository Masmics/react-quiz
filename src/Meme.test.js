import React from 'react';
import { shallow } from 'enzyme';
import Meme from './Meme';

describe('meme component', () => {
  it('renders a meme', () => {
    const wrapper = shallow(<Meme />);
    expect(wrapper).toMatchSnapshot();
  });
});
