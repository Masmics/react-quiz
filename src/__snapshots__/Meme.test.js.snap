// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`meme component renders a meme 1`] = `
<Fragment>
  <Meme
    footerText=""
    handleChange={[Function]}
  />
  <Meme
    footerText=""
    header="Meme Generator"
    image="https://images.ecosia.org/An2PbD_aWxo5_5lkuaiDNB7sl-s=/0x390/smart/http%3A%2F%2Fi0.kym-cdn.com%2Fphotos%2Fimages%2Foriginal%2F001%2F178%2F872%2F3f4.jpg"
  />
</Fragment>
`;
